overlord/snapshotstate/backend/backend.go

```diff
@@ -589,7 +593,43 @@ func writeOneSnapshotFile(targetPath string, tr io.Reader) error {
 	return nil
 }
 
-func unpackVerifySnapshotImport(r io.Reader, realSetID uint64) (snapNames []string, err error) {
+type DuplicatedSnapshotImportError struct {
+	setID uint64
+}
+
+func (e DuplicatedSnapshotImportError) Error() string {
+	return fmt.Sprintf("cannot import snapshot, already availalble as snapshot id %v", e.setID)
+}
+
+func checkDuplicatedSnapshotSetWithContentHash(ctx context.Context, contentHash []byte) error {
+	snapshotSetMap := map[uint64]client.SnapshotSet{}
+
+	// XXX: deal with import in progress here
+
+	// get all current snapshotSets
+	err := Iter(ctx, func(reader *Reader) error {
+		ss := snapshotSetMap[reader.SetID]
+		ss.Snapshots = append(ss.Snapshots, &reader.Snapshot)
+		snapshotSetMap[reader.SetID] = ss
+		return nil
+	})
+	if err != nil {
+		return fmt.Errorf("cannot calculate snapshot set hashes: %v", err)
+	}
+
+	for setID, ss := range snapshotSetMap {
+		h, err := ss.ContentHash()
+		if err != nil {
+			return fmt.Errorf("cannot calculate content hash for %v: %v", setID, err)
+		}
+		if bytes.Equal(h, contentHash) {
+			return DuplicatedSnapshotImportError{setID}
```

Comment:
```suggestion
			return &DuplicatedSnapshotImportError{setID}
```

Reply:
No strong opinion, it just felt that a struct with a single uint64 is fine to use as a value (even the fact that it's a struct is a bit debatable, probably a `type DuplicatedSnapshotImportError uint64` would be fine but it feels slightly less clear. Anyway, I am happy either way I think. Maybe @pedronis has a stronger opinion?

Reply:
Fair point, I think that's fine, I thought we use pointers for errors but we do cases similar to this already. So, +1